pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -846,4 +847,36 @@ public void validateNamespacePolicyOperation(NamespaceName namespaceName,
             }
         }
     }
+
+    public void validateTopicPolicyOperation(TopicName topicName, PolicyName policy, PolicyOperation operation) {
+        if (pulsar().getConfiguration().isAuthenticationEnabled() && pulsar().getBrokerService().isAuthorizationEnabled()) {
+            if (!isClientAuthenticated(clientAppId())) {
+                throw new RestException(Status.FORBIDDEN, "Need to authenticate to perform the request");
```

Comment:
Probably should return `Status.UNAUTHORIZED` rather than `Status.FORBIDDEN` to avoid confusion?

Reply:
Probably but the rest of this stuff content is using forbidden today.